lib/creek/styles/converter.rb

```diff
@@ -4,126 +4,30 @@ module Creek
-        when :date, :time, :date_time
-          convert_date(value, options)
-        when :bignum
-          convert_bignum(value)
-
-        ## Nothing matched
-        else
-          value
-        end
-      end
-
-      # the trickiest. note that  all these formats can vary on
-      # whether they actually contain a date, time, or datetime.
-      def self.convert_date(value, options)
-        value                        = value.to_f
-        days_since_date_system_start = value.to_i
-        fraction_of_24               = value - days_since_date_system_start
-
-        # http://stackoverflow.com/questions/10559767/how-to-convert-ms-excel-date-from-float-to-date-format-in-ruby
-        date = options.fetch(:base_date, Date.new(1899, 12, 30)) + days_since_date_system_start
-
-        if fraction_of_24 > 0 # there is a time associated
-          seconds = (fraction_of_24 * 86400).round
-          time = create_time(date) + seconds
-
-          time.year == 1899 ? time : create_date_time(time)
-        else
-          date
-        end
-      end
-
-      def self.create_time(date)
-        Time.utc(date.year, date.month, date.day)
+      BUILD_IN_CONVERTERS = [
+        Creek::Styles::ValueConvert::BigNumConvert.new,
+        Creek::Styles::ValueConvert::BoolConvert.new,
+        Creek::Styles::ValueConvert::DateTimeConvert.new,
+        Creek::Styles::ValueConvert::NumberConvert.new,
+        Creek::Styles::ValueConvert::PercentageConvert.new,
+        Creek::Styles::ValueConvert::SharedStringConvert.new,
+        Creek::Styles::ValueConvert::StringConverter.new,
+        Creek::Styles::ValueConvert::UnsupportedConverter.new,
+      ]
+
+      def initialize(custom_converters: [])
+        @_converters = custom_converters + BUILD_IN_CONVERTERS
       end
 
-      def self.create_date_time(time)
-        DateTime.civil(time.year, time.month, time.day, time.hour, time.min, time.sec)
-      end
-
-      def self.convert_bignum(value)
-        if defined?(BigDecimal)
-          BigDecimal.new(value)
-        else
-          value.to_f
+      def call(value, type, style, options = {})
+        @_converters.each do |converter|
+          if converter.is_support?(value, type, style, options)
+            return converter.convert(value, type, style, options)
+          end
         end
-      end
 
-      def self.shared_string_value(value, options)
-        return options[:shared_strings][value.to_i] unless options[:with_html]
-        node_options = {
-          :ignore_phonetic_fields => options[:ignore_phonetic_fields]
-        }
-
-        if options[:html_cell]
-          node_options[:cell_style] = options[:cell_style]
-
-          html_from(options[:shared_strings][value.to_i], node_options)
-        else
-          text_from(options[:shared_strings][value.to_i], node_options)
-        end
+        raise StandardError, "Not found converter for cell #{options[:cell_name].to_s} with type #{type.to_s} and style #{style.to_s}"
```

Comment:
You don't need to call `to_s` in interpolation
```suggestion
        raise StandardError, "Not found converter for cell #{options[:cell_name]} with type #{type} and style #{style}"
```